File: example/scalalib/script/10-custom-class/Qux.scala
//| extends: millbuild.LineCountScalaModule
//| scalaVersion: 3.8.1
package qux

def getLineCount() = {
  scala.io.Source
    .fromResource("line-count.txt")
    .mkString
}

def main() = {
  println(s"Line Count: ${getLineCount()}")
}

File: example/kotlinlib/script/10-custom-class/Qux.kt
//| extends: millbuild.LineCountKotlinModule
//| kotlinVersion: 2.2.20
package qux

fun getLineCount(): String =
    ::main.javaClass.classLoader
        .getResourceAsStream("line-count.txt")
        .readAllBytes()
        .toString(Charsets.UTF_8)

fun main() {
    println("Line Count: " + getLineCount())
}
